dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -153,7 +153,18 @@ func (cw *ClientWriter) PostRun(ctx context.Context) (err error) {
 			}
 		}
 	}
-	if err = downloader.MoveFile(src, cw.cfg.RV.RealTarget, cw.cfg.Md5); err != nil {
+
+	var expectMd5 string
+	if cw.cfg.Md5 != "" {
+		expectMd5 = cw.cfg.Md5
+	} else {
+		finishData := response.FinishData()
+		if finishData != nil {
+			expectMd5 = finishData.Md5
+		}
+	}
+
```

Comment:
I think updating p2p.cfg.Md5 in func p2p.run will be better.

```
if code == constants.CodePeerContinue {
				p2p.processPiece(response, &curItem)
			} else if code == constants.CodePeerFinish {
				p2p.finishTask(ctx, pieceWriter)
				p2p.cfg.Md5 = response.FinishData().Md5
				return nil
			}
```

Reply:
Great!